### PR TITLE
Add "instance" and "instance_id" keys to local_docker_compose return value

### LINKED ISSUE #172: encapsulate relevant information within local_docker_compose and avoid hardcoded or duplicate fixture specific info
ATM `local_docker_compose` returns a dict with only `api_key`, then code hardcodes instance id (`local-docker-tests`). Ideally all that information should come from a fixture.

I also wonder if we could may be make a "test level" fixture which would use the "session level local_docker_compose" fixture so it could also make any test specific changes, such as  `monkeypatch.setenv("DANDI_API_KEY", local_docker_compose["api_key"])`?

All that to possibly make fixture replaceable or used (later) in parametrization of the test, and to avoid duplicate code within tests using the fixture.

### PR DESCRIPTION
**Pull request by jwodder**
Also define a `local_docker_compose_env` fixture that sets `DANDI_API_KEY`.

Closes #172.

### CODE REVIEW

File: dandi/tests/test_register.py [hunk 1]
```diff
@@ -34,9 +33,8 @@ def test_smoke_metadata_present(local_docker_compose, monkeypatch, tmp_path):
     # with the given identifier
 
 
-def test_smoke_metadata_not_present(local_docker_compose, monkeypatch, tmp_path):
-    monkeypatch.setenv("DANDI_API_KEY", local_docker_compose["api_key"])
-    dandi_instance = known_instances["local-docker-tests"]
+def test_smoke_metadata_not_present(local_docker_compose_env, monkeypatch, tmp_path):
+    dandi_instance = known_instances[local_docker_compose_env["instance_id"]]
```
Comment (yarikoptic) [inline dandi/tests/test_register.py:37]: here and above
```suggestion
    dandi_instance = local_docker_compose_env["instance"]
```
?

File: dandi/tests/test_upload.py [hunk 2]
```diff
@@ -12,14 +12,13 @@
...
-def test_upload(local_docker_compose, monkeypatch, tmp_path):
+def test_upload(local_docker_compose_env, monkeypatch, tmp_path):
     DIRNAME1 = "sub-anm369963"
     FILENAME1 = "sub-anm369963_ses-20170228.nwb"
     DIRNAME2 = "sub-anm372793"
     FILENAME2 = "sub-anm372793_ses-20170508.nwb"
 
-    monkeypatch.setenv("DANDI_API_KEY", local_docker_compose["api_key"])
-    dandi_instance = "local-docker-tests"
+    dandi_instance = local_docker_compose_env["instance_id"]
```
Comment (yarikoptic) [inline dandi/tests/test_upload.py:21]: here and in another test below (more code would need tune up for the rename):

```suggestion
    dandi_instance_id = local_docker_compose_env["instance_id"]
```
for consistency?

File: dandi/tests/test_register.py (hunk 1 above)
Comment (jwodder) [reply dandi/tests/test_register.py]: Done.

File: dandi/tests/test_upload.py (hunk 2 above)
Comment (jwodder) [reply dandi/tests/test_upload.py]: Done.

File: dandi/tests/test_upload.py [hunk 3]
```diff
@@ -35,43 +34,42 @@ def test_upload(local_docker_compose, monkeypatch, tmp_path):
         metadata = yaml.safe_load(fp)
     dandi_id = metadata["identifier"]
 
-    client = girder.get_client(known_instances[dandi_instance].girder)
+    client = girder.get_client(known_instances[dandi_instance_id].girder)
```
Comment (yarikoptic) [inline dandi/tests/test_upload.py:37]: here and similarly in the other tests:

```suggestion
    client = girder.get_client(local_docker_compose_env["instance"].girder)
```

so we could avoid using `known_instances` as much as possible and take advantage of encapsulation of all relevant information within the fixture.

Comment (jwodder) [reply dandi/tests/test_upload.py]: Done.

**Comment by yarikoptic** [comment]
AWESOME! Thank you!

**Pull request merged**